Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,196 @@
+            let smallerSide = frame.size.width < frame.size.height ? frame.size.width : frame.size.height
+            let scale = smallerSide/imgQR.extent.size.width
+            let transformedImage = imgQR.transformed(by: CGAffineTransform(scaleX: scale, y: scale))
+            let imageQRCode = UIImage(ciImage: transformedImage)
+
+            return imageQRCode
+        }
+        return nil
+    }
+
+    @IBAction func onClose(_ sender: Any) {
+        self.dismissView()
+    }
+
+    @IBAction func showInfo(_ sender: Any) {
+        DispatchQueue.main.async {
+            let alertController = UIAlertController(title: LocalizedStrings.receiveDCR,
+                                                    message: LocalizedStrings.receiveInfo,
+                                                    preferredStyle: UIAlertController.Style.alert)
+            alertController.addAction(UIAlertAction(title: LocalizedStrings.gotIt,
+                                                    style: UIAlertAction.Style.default,
+                                                    handler: nil))
+            self.present(alertController, animated: true, completion: nil)
+        }
+    }
+
+    func updateSelectedAccount(_ selectedWalletId: Int, _ selectedAccount: DcrlibwalletAccount) {
+        guard let wallet = WalletLoader.shared.multiWallet.wallet(withID: selectedWalletId) else {
+            return
+        }
+        self.selectedWallet = wallet
+        self.selectedAccount = selectedAccount
+
+        self.walletNameLabel.text = wallet.name
+        self.accountNameLabel.text = selectedAccount.name
+
+        let totalBalanceRoundedOff = (Decimal(selectedAccount.dcrTotalBalance) as NSDecimalNumber).round(8)
+        self.totalAccountBalanceLabel.attributedText = Utils.getAttributedString(str: "\(totalBalanceRoundedOff)", siz: 15.0, TexthexColor: UIColor.appColors.darkBlue)
+        self.updateWalletAddressAndQRCode(receiveAddress: wallet.currentAddress(selectedAccount.number, error: nil))
+    }
+
+    @IBAction func showMenu(_ sender: Any) {
+        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        let cancelAction = UIAlertAction(title: LocalizedStrings.cancel, style: .cancel, handler: nil)
+
+        let generateNewAddressAction = UIAlertAction(title: LocalizedStrings.genNewAddr, style: .default, handler: { (alert: UIAlertAction!) -> Void in
+            self.generateNewAddress()
+        })
+
+        alertController.addAction(cancelAction)
+        alertController.addAction(generateNewAddressAction)
+
+        self.present(alertController, animated: true, completion: nil)
+    }
+
+    @IBAction func onShare(_ sender: Any) {
+        var img: UIImage = self.addressQRCodeImageView.image!
+
+        if img.cgImage == nil {
+            guard let ciImage = img.ciImage, let cgImage = CIContext(options: nil).createCGImage(ciImage, from: ciImage.extent) else {return}
```

Comment:
We already checked that `img.ciImage` is nil, this guard will always fail. Please refactor.